### PR TITLE
Feat/forms

### PR DESCRIPTION
**Pull request by over-geek**
## Pull Request Summary

This pull request introduces several enhancements and bug fixes to the blog application. The main changes include:

1. **Form Partial for Posts and Comments:**
   - Created partials for forms related to posts and comments to promote code reusability.
   - Modified the views to use these partials.

2. **Current User Method:**
   - Added a `current_user` method in the ApplicationController to make the current user data available to all controllers.

3. **Likes Functionality:**
   - Implemented the ability for users to add likes to posts.
   - Created a `LikesController` with a `create` action.
   - Added a "Like" button in the post show view.

4. **Redirection Fixes:**
   - Updated the redirection logic in the `LikesController` to ensure users are redirected to the correct post after adding a like.

5. **Error Handling:**
   - Improved error handling in the `create` actions of Posts, Comments, and Likes controllers.
   - Displayed flash messages for success and error scenarios.

6. **Refactoring:**
   - Refactored parts of the code for better readability and maintainability.

7. **Other Fixes:**
   - Fixed issues related to the creation of comments and posts.

## How to Test

1. Clone this branch locally.
2. Run database migrations (`rails db:migrate`) if needed.
3. Start the Rails server (`rails server`).
4. Navigate to the appropriate routes (e.g., create a post, add a comment, like a post) and ensure functionality works as expected.

 Feedback and suggestions are welcome!

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -11,17 +13,28 @@ def show
   end
 
   def new
+    @user = User.find(params[:user_id])
     @post = Post.new
   end
 
   def create
-    @post = current_user.posts.new(post_params)
+    @post = @user.posts.new(post_params)
```
Comment (LionRouge1) [inline app/controllers/posts_controller.rb]: - [ ] Nice work implementing the create action for the post :clap: However, according to the project requirement post should be created on behalf of the `current_user` (using the method that you created in your ApplicationController). Therefore I suggest that you change the `@user` to use the `current_user` method created in the ApplicationController

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
   def new
     @post = Post.find(params[:post_id])
+    @user = @post.author
     @comment = Comment.new
   end
 
   def create
     @post = Post.find(params[:post_id])
     @comment = Comment.new(comment_params.merge(post: @post))
+    @comment.author = @post.author
```
Comment (LionRouge1) [inline app/controllers/comments_controller.rb]: - [ ] Nice work implementing the create action for the comment :clap: However, according to the project requirement comment should be created on behalf of the `current_user` (using the method that you created in your ApplicationController). Therefore I suggest that you update the comment's author to use the `current_user` method created in the ApplicationController

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @over-geek, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Create a method called `current_user` in your ApplicationController :heavy_check_mark: 
- Create a form to create post and comment :heavy_check_mark: 
- Create post and comment on behalf of the logged in user :x:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @over-geek, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**